aws/s3/s3_integration_test.go

```diff
@@ -0,0 +1,1002 @@
+
+	err = client.Get(TestBucket, testDataFilepath, "", &dataObject)
+
+	// ASSERT
+
+	// obtained object just fine
+	assert.Nil(t, err)
+
+	// object is what we expect
+	testFileContents, _ := os.ReadFile(testDataFilepath)
+	assert.Equal(t, testFileContents, dataObject.Bytes())
+}
+
+func TestS3GetWithLastModified(t *testing.T) {
+	const (
+		CustomAWSEndpointURL = "http://localhost:4566"
+		AWSRegion            = "ap-southeast-2"
+		TestBucket           = "test-bucket"
+	)
+
+	// ARRANGE
+
+	// setup environment variables to access LocalStack
+	os.Setenv("AWS_REGION", AWSRegion)
+	os.Setenv("AWS_SECRET_ACCESS_KEY", "test")
+	os.Setenv("AWS_ACCESS_KEY_ID", "test")
+	os.Setenv("CUSTOM_AWS_ENDPOINT_URL", CustomAWSEndpointURL)
+
+	// create bucket
+	cmd := exec.Command(
+		"aws", "s3api",
+		"create-bucket",
+		"--bucket", TestBucket,
+		"--create-bucket-configuration", fmt.Sprintf(
+			"{\"LocationConstraint\": \"%v\"}", AWSRegion),
+		"--endpoint-url", CustomAWSEndpointURL)
+
+	defer exec.Command(
+		"aws", "s3",
+		"rb", fmt.Sprintf("s3://%v", TestBucket),
+		"--force",
+		"--endpoint-url", CustomAWSEndpointURL).Run()
+
+	out, err := cmd.CombinedOutput()
+	require.Nil(t, err, fmt.Sprintf("Command failed: %v", string(out)))
+
+	// create test data
+	testDataFilepath := filepath.Join(t.TempDir(), "data.txt")
+	testFile, _ := os.Create(testDataFilepath)
+	testFile.WriteString("hello world")
+	testFile.Close()
+
+	// populate bucket
+	cmd = exec.Command(
+		"aws", "s3api",
+		"put-object",
+		"--bucket", TestBucket,
+		"--key", testDataFilepath,
+		"--body", testDataFilepath,
+		"--endpoint-url", CustomAWSEndpointURL)
+
+	out, err = cmd.CombinedOutput()
+	require.Nil(t, err, fmt.Sprintf("Command failed: %v", string(out)))
+
+	// get last modified
+	cmd = exec.Command(
+		"aws", "s3api",
+		"head-object",
+		"--bucket", TestBucket,
+		"--key", testDataFilepath,
+		"--endpoint-url", CustomAWSEndpointURL)
+
+	out, err = cmd.CombinedOutput()
+	require.Nil(t, err, fmt.Sprintf("Command failed: %v", string(out)))
+
+	var metaData map[string]interface{}
+	json.Unmarshal(out, &metaData)
+	testLastModified, err := time.Parse(
+		"2006-01-02T15:04:05", metaData["LastModified"].(string)[0:19])
+	require.Nil(t, err, fmt.Sprintf("Last modified test setup failed: %v", err))
+
+	// create client
+	client, err := New()
+	require.Nil(t, err, fmt.Sprintf("Error creating s3 client: %v", err))
+
+	// ACTION
+
+	dataObject := bytes.Buffer{}
+
+	lastModified, err := client.GetWithLastModified(
+		TestBucket, testDataFilepath, "", &dataObject)
+
+	// ASSERT
+
+	// obtained object just fine
+	assert.Nil(t, err)
+
+	// object is what we expect
+	testFileContents, _ := os.ReadFile(testDataFilepath)
+	assert.Equal(t, testFileContents, dataObject.Bytes())
+
+	// modified time is what we expect
+	assert.Equal(t, testLastModified, lastModified)
+
+	// bonus test, LastModified function
+	lastModified, err = client.LastModified(TestBucket, testDataFilepath, "")
```

Comment:
If it's unlikely to return error so we're not checking it, then use `last modifier, _ = client.LastModified...`, otherwise error should be checked.